src/redux/optimismExplorer.js

```diff
@@ -83,30 +82,28 @@ export const optimismExplorerInit = () => async (dispatch, getState) => {
       return;
     }
 
-    const tokenAddresses = assets.map(
-      ({ asset: { asset_code } }) => asset_code
+    const tokenAddresses = assets.map(({ asset: { asset_code } }) =>
+      toLower(asset_code)
     );
 
     const balances = await fetchAssetBalances(
-      assets.map(({ asset: { asset_code } }) => asset_code),
+      tokenAddresses,
       accountAddress,
       network
     );
 
-    let total = BigNumber.from(0);
-
+    let updatedAssets = assets;
     if (balances) {
-      Object.keys(balances).forEach(key => {
-        for (let i = 0; i < assets.length; i++) {
-          if (assets[i].asset.asset_code.toLowerCase() === key.toLowerCase()) {
-            assets[i].quantity = balances[key];
-            break;
-          }
-        }
-        total = total.add(balances[key]);
+      updatedAssets = assets.map(assetAndQuantity => {
```

Comment:
this is where the fix actually happens: doing a mapping that returns a new object with quantity updated